WordPress/gutenberg
# New block: Post navigation

## What?


Closes https://github.com/WordPress/gutenberg/issues/55157

The purpose of adding this new block is to:

- Make it easier for users to add both the next and the previous posts links together, instead of having to insert two separate blocks and (commonly) wrap them inside a group/row/stack/column to position them.
-  Improve the accessibility of the post navigation

## Why?

- Navigations should be wrapped inside a `<nav> `element.
- Needing to add the next and previous posts blocks separately is time consuming.
- The most common use case is to have both a previous and next post link in your template, not just one of them.
- It makes the post navigation work more like the query pagination, which already uses a wrapper.

## How?

Adds a new block called Post Navigation.
The block adds a `<nav>` element and inserts two post navigation link blocks as inner blocks.
One of the inner blocks is of the "previous" block variation.
The block supports adding an aria label. The user can add a "navigation name" in the Advanced panel.
The block also supports color and typography options.

##  Note on accessibility
While using a nav element is an improvement, this block is still not as accessible as the post navigation function that is available for classic themes, because it does not have a visually hidden heading inside the nav element, above the links.
[the_post_navigation](https://developer.wordpress.org/reference/functions/the_post_navigation/)
[get_the_post_navigation](https://developer.wordpress.org/reference/functions/get_the_post_navigation/)
[navigation markup](https://developer.wordpress.org/reference/functions/_navigation_markup/)

## Testing Instructions
Activate a block theme.
The post navigation block can be tested using a single post, or the single post template.
Add the block.

Confirm that:
- Both the next and previous post navigation links show in the editor.
- The toolbar options work.
- The setting panel options work.
- The blocks and their settings display correctly on the front.

### Testing Instructions for Screen readers

Activate a block theme.
The post navigation block can be tested using a single post, or the single post template.
Add the block.
Save and go to the front of the website.
Navigate to where you placed the block.
Confirm that the first of the two links are announced as part of a navigation, and that the aria label is announced.
If you did not change the aria label in the blocks Advanced panel, the default label is "Posts".

## Screenshots or screencast 
<img width="1512" alt=" The Post Navigation block in the Site Editor, with the list view and settings panel opened." src="https://github.com/WordPress/gutenberg/assets/7422055/145c7a7f-fedd-40c9-8e98-b41ddb2a496a">
